parity/lib.rs

```diff
@@ -141,6 +141,37 @@ fn print_hash_of(maybe_file: Option<String>) -> Result<String, String> {
 	}
 }
 
+fn run_deadlock_detection_thread() {
+	#[cfg(feature = "deadlock_detection")]
```

Comment:
What is the reason of putting this `conditional compilation flag` inside the function rather than outside?